src/lang-go.ts

```diff
@@ -189,31 +190,22 @@ async function tryToCreateAccessToken(): Promise<string | undefined> {
 }
 
 async function connectAndInitialize(
-    address: string,
+    address: URL,
     root: URL,
     token: string | undefined
 ): Promise<rpc.MessageConnection> {
-    const connectingToGoLangserverHelp = [
-        `Unable to connect to the Go language server at ${address}.`,
-        `Make sure ${'go.address' as keyof Settings} in your Sourcegraph settings is set to the address of the language server (e.g. wss://sourcegraph.example.com/go).`,
-        `Read the full documentation for more information: https://github.com/sourcegraph/sourcegraph-go`,
-    ].join('\n')
-
-    const connectingToSourcegraphHelp = [
-        `The Go language server running on ${address} was unable to fetch repository contents from Sourcegraph running on ${sourcegraphURL()}.`,
-        `Make sure ${'go.sourcegraphUrl' as keyof Settings} in your settings is set to the address of Sourcegraph from the perspective of the language server (e.g. http://sourcegraph-frontend:30080 when running in Kubernetes).`,
-        `Read the full documentation for more information: https://github.com/sourcegraph/sourcegraph-go`,
-    ].join('\n')
-
-    const connection = (await new Promise((resolve, reject) => {
+    const connection = (await new Promise<wsrpc.MessageConnection>((resolve, reject) => {
```

Comment:
This should use an async IIFE instead of the Promise constructor (Promise constructor should be avoided where possible, and it's possible here)